platform/database/tests/database_test.go

```diff
@@ -0,0 +1,281 @@
+	}
+
+	if getTestData.ID != firstUUID || getTestData.Value != firstValue {
+		t.Fatalf("expected test data with ID : %s and Value: %s, but got %#v", firstUUID, firstValue, getTestData)
+	}
+
+	db.Close()
+
+	// recreating a database each time to prove it can persist across different connections
+	db, err = database.Connect(cfg)
+	if err != nil {
+		t.Fatalf("could not connect to the database: %#v", err)
+	}
+
+	var selectTestData []testData
+	err = db.SelectContext(
+		context.Background(),
+		&selectTestData,
+		"SELECT * from test_data",
+	)
+
+	if err != nil {
+		t.Fatalf("could not SelectContext: %#v", err)
+	}
+
+	checkTestData := map[string]bool{
+		"first_row":  false,
+		"second_row": false,
+	}
+
+	for _, row := range selectTestData {
+		if row.ID == firstUUID && row.Value == firstValue && checkTestData["first_row"] == false {
+			checkTestData["first_row"] = true
+		} else if row.ID == secondUUID && row.Value == secondValue && checkTestData["second_row"] == false {
+			checkTestData["second_row"] = true
+		} else {
+			t.Fatalf("Expected selectTestData to be first test data and second test data but got %#v", selectTestData)
+		}
+	}
+
+	if !checkTestData["first_row"] {
+		t.Fatalf("expect selectTestData to carry firstValue but got %#v", selectTestData)
+	}
+
+	if !checkTestData["second_row"] {
+		t.Fatalf("expect selectTestData to carry secondValue but got %#v", selectTestData)
+	}
+
+	db.Close()
+
+	// recreating a database each time to prove it can persist across different connections
+	db, err = database.Connect(cfg)
+	if err != nil {
+		t.Fatalf("could not connect to the database: %#v", err)
+	}
+
+	tx, err := db.Begin()
+
+	if err != nil {
+		t.Fatalf("could not start a transaction: %#v", err)
+	}
+	defer func() { tx.Rollback() }()
+
+	_, err = tx.ExecContext(context.Background(), "UPDATE test_data SET value = $1 WHERE id = $2", "updated_value", firstUUID)
+	if err != nil {
+		t.Fatalf("could not ExecContext in a transaction: %#v", err)
+	}
+
+	// This is inserting a primary key already existing in order to corrupt the transaction
+	_, err = tx.ExecContext(
+		context.Background(),
+		`INSERT INTO test_data (id, value) VALUES ($1, $2);`,
+		firstUUID,
+		firstValue,
+	)
+
+	if err == nil {
+		t.Fatalf("ExecContext should return an error")
+	}
+
+	err = tx.Commit()
+	if err == nil {
+		t.Fatalf("should not be able to commit faulty transaction")
+	}
+
+	err = db.Close()
+	if err != nil {
+		t.Fatalf("COULD NOT CLOSE THE DATABASE AAAAAAAAAAAH: %v", err)
+	}
+
+	db, err = database.Connect(cfg)
+	if err != nil {
+		t.Fatalf("could not connect to database: %v", err)
+	}
+
+	tx, err = db.Begin()
+
+	err = db.GetContext(
+		context.Background(),
+		&getTestData,
+		`SELECT * from test_data WHERE ID = $1`,
+		firstUUID,
+	)
+	if err != nil {
+		t.Fatalf("could not use getContext %#v", err)
+	}
+
+	if getTestData.Value != firstValue {
+		t.Fatalf("A part of the rollbacked transaction has been committed")
+	}
```

Comment:
I don't think this check is very valuable to be honest. On line 199 you are trying to insert a new row with the same ID and value of the very first row you inserted, which should throw an error because of the primary key ofcourse. 

Let's say that for some f*d up reason that statement made an update to that previously inserted row and changed the value of the row to the value passed in to the statement on line 199. Or that it inserted as well because the primary key was not set on the table.

This check would never catch it because you are only trying to fetch one and were trying  to insert the exact same data on line 199, so any updates would never be noticed in this test because of the dataset you used.

So I would suggest to 1) try to fetch a list and ensure only 1 is returned 2) ensure that on line 199 you insert a different value (from the existing value for the firstUUID) for the value (see the confusion here 😉)